lightning/src/routing/router.rs

```diff
@@ -1446,14 +1446,31 @@ where L::Target: Logger {
 
 	// Draw multiple sufficient routes by randomly combining the selected paths.
 	let mut drawn_routes = Vec::new();
-	for i in 0..payment_paths.len() {
+	let mut prng = ChaCha20::new(random_seed_bytes, &[0u8; 12]);
+	let mut random_index_bytes = [0u8; ::core::mem::size_of::<usize>()];
+
+	let num_permutations = payment_paths.len();
+	let mut seen_permutation_scids: HashSet<Vec<Vec<u64>>> = HashSet::new();
```

Comment:
Well not tracking them might result in duplicate permutations, which could reduce the variety of inputs for the selection logic down the line. We could work around this by simply generating more permutations, but this will of course will increase the runtime. 
 
However, since we're going to sort the candidates shortly anyways, the variety might even be more important than the random order of the input data? So if we wanted to make this simpler, just shuffling `payment_paths` once before and then sticking with the prior 'wrap-around slice` approach to generate unique (although sequential) permutations could be an option?

Reply:
Alright, with https://github.com/lightningdevkit/rust-lightning/pull/1359/commits/bac11886a3c6aedebf5e27d7e88e8d2d428c5d04 I don't track duplicates anymore. 
For now, I left the number of permutations equal to the number of payment path candidates.